containers/java-postgres/README.md

```diff
@@ -0,0 +1,146 @@
+
+```yaml
+build:
+  context: ..
+  dockerfile: .devcontainer/Dockerfile
+    args:
+      # Update 'VARIANT' to pick an version of Java: 11, 17.
+      # Append -bullseye or -buster to pin to an OS version.
+      # Use -bullseye variants on local arm64/Apple Silicon.
+      VARIANT: 11-bullseye
+```
+
+You also can connect to PostgreSQL from an external tool when using VS Code by updating `.devcontainer/devcontainer.json` as follows:
+
+```json
+"forwardPorts": [ "5432" ]
+```
+
+### Adding another service
+
+You can add other services to your `docker-compose.yml` file [as described in Docker's documentation](https://docs.docker.com/compose/compose-file/#service-configuration-reference). However, if you want anything running in this service to be available in the container on localhost, or want to forward the service locally, be sure to add this line to the service config:
+
+```yaml
+# Runs the service on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+network_mode: service:db
+```
+
+### Debug Configuration
+
+Note that only the integrated terminal is supported by the Remote - Containers extension. You may need to modify `launch.json` configurations to include the following value if an external console is used.
+
+```json
+"console": "integratedTerminal"
+```
+
+### Installing Maven or Gradle
+
+You can opt to install a version of Maven or Gradle by adding `INSTALL_MAVEN: "true"` or `INSTALL_GRADLE: "true"` to build args in `.devcontainer/docker-compose.yml`. Both of these are set by default. For example:
+
+```yaml
+args:
+  VARIANT: 11
+  INSTALL_GRADLE: "true"
+  INSTALL_MAVEN: "true"
+```
+
+Remove the appropriate arg or set its value to `"false"` to skip installing the specified tool.
+
+You can also specify the version of Gradle or Maven if needed.
+
+```yaml
+args:
+  VARIANT: 11
+  INSTALL_GRADLE: "true"
+  MAVEN_VERSION: "3.8.3"
+  INSTALL_MAVEN: "true"
+  GRADLE_VERSION: "7.2"
+```
+
+### Installing Node.js
+
+Given JavaScript front-end web client code written for use in conjunction with a Java back-end often requires the use of Node.js-based utilities to build, this container also includes `nvm` so that you can easily install Node.js. You can enable installation and change the version of Node.js installed or disable its installation by updating the `args` property in `.devcontainer/docker-compose.yml`.
+
+```yaml
+args:
+  VARIANT: 11
+  NODE_VERSION: "10" # Set to "none" to skip Node.js installation
+```
+
+### Adding the definition to your folder
+
+1. If this is your first time using a development container, please see getting started information on [setting up](https://aka.ms/vscode-remote/containers/getting-started) Remote-Containers or [creating a codespace](https://aka.ms/ghcs-open-codespace) using GitHub Codespaces.
+
+2. Start VS Code and open your project folder or connect to a codespace.
```

Comment:
I would propose cutting this section for the most part and instead referring to the existing public web docs. It's possible that the user flows would change over time and make this documentation go out of date.

Reply:
I would be against this. Dan and I have been trying to make this document additive instead of subtractive, and we don't want it to state anything different from the current documentation. 

In other words, this is copied from the existing java container.

Reply:
If you've considered it and decided against it, I'm okay with that resolution